Apps/UnitTests/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+if((NOT WIN32) OR WINDOWS_STORE)
+    message(FATAL_ERROR "Unsupported platform: ${CMAKE_SYSTEM_NAME}")
+endif()
+
+include(FetchContent)
+FetchContent_Declare(
+  googletest
+  # Specify the commit you depend on and update it regularly.
+  URL https://github.com/google/googletest/archive/23ef29555ef4789f555f1ba8c51b4c52975f0907.zip
+)
+# For Windows: Prevent overriding the parent project's compiler/linker settings
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
+
+add_executable(UnitTests "UnitTests.cpp")
+
+target_compile_definitions(UnitTests PRIVATE UNICODE)
+target_compile_definitions(UnitTests PRIVATE _UNICODE)
+
+target_include_directories(UnitTests PRIVATE "Source" ".")
+
+target_link_libraries(UnitTests
+    gtest_main
+    UrlLib)
+
+
+add_test(NAME UnitTests COMMAND UnitTests)
+
+set_property(TARGET UnitTests PROPERTY FOLDER Apps)
```

Comment:
Probably good to add the googletest projects to a folder also, so they show up in the dependencies folder like other dependencies.

Reply:
Agreed, but I'm not sure how to put content from FetchContent into a subdirectory. I will keep looking into this

Reply:
This `set_property(TARGET UnitTests PROPERTY FOLDER Apps)` just puts the UnitTests project in a solution folder (or filter) in Visual Studio. I would expect you can do the same for the googletest projects.

Reply:
Yep, I was definitely way overthinking this  :man_facepalming: The GoogleTest projects now all go in Dependencies\GoogleTest